File: 7 Jan 2026 Count distinct elements in every window/code.py
class Solution:
    def countDistinct(self, arr, k):
        res=[]
        mp={}
        n,j=len(arr),0
        for i in range(0,n):
            if((i-j+1)>k):
                mp[arr[j]]-=1
                if(mp[arr[j]]==0):
                    del mp[arr[j]]
                j+=1
            mp[arr[i]]=mp.get(arr[i],0)+1
            if((i-j+1)==k):
                res.append(len(mp))
            
        return res
        
        

File: 13 Jan 2026 Bus Ticket Change/code.cpp
class Solution {
  public:
    bool canServe(vector<int> &arr) {
        int f=0,t=0;
        for(int i:arr){
            if(i==5)f++;
            else if(i==10){
                if(f)f--,t++;
                else return false;
            }
            else{
                if(t and f){
                    t--,f--;
                }
                else if(f>2){
                    f-=3;
                }
                else return false;
            }
        }
        return true;
    }
};